rx-extensions/src/main/java/com/appunite/rx/subjects/CacheSubject.java

```diff
@@ -67,13 +72,15 @@
         };
     }
 
-    private CacheSubject(@Nonnull final CacheCreator<T> cacheCreator, DelegateOnSubscribe<T> delegateOnSubscribe) {
+    private CacheSubject(@Nonnull final CacheCreator<T> cacheCreator,
+                         @Nonnull DelegateOnSubscribe<T> delegateOnSubscribe,
+                         final boolean skipFirstNull) {
         super(delegateOnSubscribe);
         delegateOnSubscribe.setDelegate(new OnSubscribe<T>() {
             @Override
             public void call(final Subscriber<? super T> child) {
                 final T t = cacheCreator.readFromCache();
-                if (t != null) {
+                if (!skipFirstNull || t != null) {
```

Comment:
No.. but previously null means cache is clear so we should skip it when subscribing. if we set skipFirstNull to false null mean that in cache is null ;)